Engine/DataFeeds/Enumerators/OpenInterestEnumerator.cs

```diff
@@ -0,0 +1,139 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using NodaTime;
+using QuantConnect.Data;
+using QuantConnect.Data.Market;
+
+namespace QuantConnect.Lean.Engine.DataFeeds.Enumerators
+{
+    /// <summary>
+    /// Aggregates open interest bars ready to be time synced
+    /// </summary>
+    public class OpenInterestEnumerator : IEnumerator<BaseData>
+    {
+        private readonly TimeSpan _barSize;
+        private readonly DateTimeZone _timeZone;
+        private readonly ITimeProvider _timeProvider;
+        private readonly ConcurrentQueue<OpenInterest> _queue;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OpenInterestEnumerator"/> class
+        /// </summary>
+        /// <param name="barSize">The OI bar size to produce</param>
+        /// <param name="timeZone">The time zone the raw data is time stamped in</param>
+        /// <param name="timeProvider">The time provider instance used to determine when bars are completed and
+        /// can be emitted</param>
+        public OpenInterestEnumerator(TimeSpan barSize, DateTimeZone timeZone, ITimeProvider timeProvider)
+        {
+            _barSize = barSize;
+            _timeZone = timeZone;
+            _timeProvider = timeProvider;
+            _queue = new ConcurrentQueue<OpenInterest>();
+        }
+        /// <summary>
+        /// Pushes the tick into this enumerator. This tick will be aggregated into a OI bar
+        /// and emitted after the alotted time has passed
+        /// </summary>
+        /// <param name="data">The new data to be aggregated</param>
+        public void ProcessData(Tick data)
+        {
+            OpenInterest working;
+
+            if (!_queue.TryPeek(out working))
```

Comment:
What about the else condition here? Won't we be dropping data?

Also, the class doc states that this performs aggregation, perhaps the else condition should perform aforementioned aggregation? or update the doc to be explicit about this type's behavior.